### DISCUSSION TITLE
Run tool for each chat message

### DISCUSSION THREAD

**Discussion by serge-dziuba** [FlowiseAI/Flowise#2298] [Q&A]
Hi, I'm trying to implement logging customer intractions in 3rd party system. 
I'm using Custom Tool with OpenAI Tool Agent agent (gpt-3.5-turbo).
My custom tool should be run for each incoming message in a chat, but flowise just ignore my prompts in most of times. Also it never runs my tool with other custom tools simultaneously.

Some tool descriptions I've tried:
- Use this tool for logging customer messages
- Run this tool for each customer message
- Run this tool when customer starts a conversation
- Log each customer message

Also my Tool Agent description contains prompt: "You should log each user message using `log_customer_visit` tool."

Any advice?

**Comment by Jaredude** [comment]
Running "custom tools simultaneously" as you stated is not something any of the current models do well, if at all.

If you want to perform custom logging, you would need to include custom logging code with each of the custom tools. Alternative, you could modify the existing chat log code so that it performs whatever custom logging you need to perform.
